### ISSUE TITLE
Integrate with treesit.el?

### ISSUE THREAD

**Issue by aspiers** [Fanael/rainbow-delimiters#81]
Now that treesit.el is part of core emacs, are there any plans to integrate rainbow-delimiters with it?  Treesitter can correctly parse many languages, so it can deal with countless corner cases which I'm guessing rainbow-delimiters can't currently.

**Comment by magthe** [comment]
Yes, maybe that might also be a way of making rainbow-delimeters usable with Haskell. The following will lead it to wrongly think that there are unmatched parens

```haskell
foo = \(a, b) -> (b, a)
```

(That's how the standard formatting tools format a lambda accepting a tuple.)
